### DISCUSSION TITLE
Strange ride rating bonus for path in first slot

### DISCUSSION THREAD

**Discussion by henke37** [OpenRCT2/OpenRCT2#15729] [Feature requests] [answered]
The ride rating computation code gives out bonuses for paths adjacent to the track of the ride. It is strangely split into bonuses for "first slot" and "the other slots". This makes no sense from a gameplay perspective.

Instead, what would make sense is to split the bonus based on if it is a normal path or a queue area.

I suspect that this is a result of the comparison getting mangled in some refactoring. d25856d59c6c20edf638b24c1bba86771c5cfb54 seems like a commit where the queue status is indeed checked.

**Comment by duncanspumpkin** [comment]
It makes no sense but that is what the original game did. We have tried to keep things as close to original. It certainly could be changed though.

**Comment by henke37** [comment]
Can someone confirm that's what actually happened in RCT1, before the expansions messed with the paths? Because this has to be a bug, be it in the expansions or in openrct.

**Comment by duncanspumpkin** [comment]
I can't easily check RCT1 you will have to just run some tests and see what it does.

**Comment by henke37** [comment]
Took me some time to check it, due to my original copy being busted. However, I was able to test in a demo version.

It is indeed the queue path that gives this bonus and no other path types.

I think the original cause of this mess is the reworking of the path tool panel, in the original version the queue path is on the left.

![SCR1](https://user-images.githubusercontent.com/4971363/90561671-5a592f80-e1a1-11ea-9166-0fabd4d6562b.PNG)

**Comment by henke37** [comment]
Can I get the "original bug" label?

**Comment by duncanspumpkin** [comment]
Is it really a bug though. It's an oddity for sure.

**Comment by henke37** [comment]
I think it's a bug that the behavior changed from vanilla RCT 1.0. This was intended to be for the queues, but it broke in the path panel redesign in the expansions.

**Comment by IntelOrca** [comment]
I think we have changed the logic for this now in NSF, but I can't remember how we changed it.

**Comment by henke37** [comment]
Reading the logic suggests that it's intentionally maintaining the bug now. It's explicitly tracking the legacy slot index for every path tile.

**Comment by Gymnasiast** [comment, marked as answer]
I checked, and the original RCT1 without expansions indeed uses slot 0 for the queue, while the other parts start at index 4. AA introduced more types of queue, with indexes 1, 2 and 3.

It has been broken since either RCT2 or even Added Attractions, depending on how RCT1 did this check: RCT1 used the lower two bits of the path index as a colour, so it might just have checked for `(path_index & 0b11100) == 0`).
